ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosDataAccessor.java

```diff
@@ -292,7 +304,16 @@ void deleteMetadata(List<CloudBlobMetadata> blobMetadataList) throws CosmosExcep
       }
     } else {
       for (CloudBlobMetadata metadata : blobMetadataList) {
-        deleteMetadata(metadata);
+        try {
+          deleteMetadataAsync(metadata).join();
```

Comment:
I think we should have a list of CompletableFuture and then after the for loop, we can wait on all of them. Blocking on each delete defeat the purpose of async operations.